src/allowance/utils.ts

```diff
@@ -9,13 +10,19 @@ import { fetchDataUsingMulticall, MultiCallData } from '../utils/multicall'
 export const getApproved = async (
   signer: Signer,
   tokenAddress: string,
-  contractAddress: string
+  contractAddress: string,
+  transactionRequest?: TransactionRequest
 ): Promise<BigNumber> => {
   const signerAddress = await signer.getAddress()
   const erc20 = new Contract(tokenAddress, ERC20_ABI, signer) as ERC20Contract
 
   try {
-    const approved = await erc20.allowance(signerAddress, contractAddress)
+    const approved = await erc20.allowance(signerAddress, contractAddress, {
+      gasLimit: transactionRequest?.gasLimit,
+      gasPrice: transactionRequest?.gasPrice,
+      maxFeePerGas: transactionRequest?.maxFeePerGas,
+      maxPriorityFeePerGas: transactionRequest?.maxPriorityFeePerGas,
+    })
```

Comment:
Can we pass the whole object here like 

`const approved = await erc20.allowance(signerAddress, contractAddress, transactionRequest)`

Reply:
I only want to pick these gas related changes that the developer modifies.
Hence avoiding the usage of whole object.

There are variables like: `data` and `customData` which I assumed shouldn't be modified. Although the contract would already have these fields filled with the ongoing interaction.